### PR TITLE
Lint about `format!("string lit")` with no argument

### LINKED ISSUE #438: Superfluous `format!`
E.g. code like this

```
hello(&format!("simple string"));
```

can always be rewritten as

```
hello("simple string")
```

### PR DESCRIPTION
**Pull request by mcarton**
Fix  #438. Clippy was “affected” three times BTW.
Also update the module list in `update_lints.py` because I *always* forget that one :blush:.

### CODE REVIEW

**Comment by mcarton** [comment]
Oh wait, it should also lint [`format!("{}", foo)`](https://github.com/Manishearth/rust-clippy/pull/645#discussion_r52376966).

File: src/format.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+use rustc::lint::*;
+use rustc_front::hir::*;
+use utils::{is_expn_of, span_lint};
+
+/// **What it does:** This lints about use of `format!("string literal with no argument")`.
+///
+/// **Why is this bad?** There is no point of doing that. If you want a `String` you can use
+/// `to_owned` on the string literal. The even worst `&format!("foo")` is often encountered in the
```
Comment (llogiq) [inline src/format.rs]: > even worst

I believe this should read 'even worse'.

File: src/types.rs [hunk 2]
```diff
@@ -491,12 +491,12 @@ fn check_type(cx: &LateContext, ty: &Ty) {
...
         visitor.score
     };
-    // println!("{:?} --> {}", ty, score);
+
     if score > 250 {
         span_lint(cx,
                   TYPE_COMPLEXITY,
                   ty.span,
-                  &format!("very complex type used. Consider factoring parts into `type` definitions"));
+                  &"very complex type used. Consider factoring parts into `type` definitions");
```
Comment (Manishearth) [inline src/types.rs]: you don't need an `&` sign there.

File: util/update_lints.py [hunk 3]
```diff
@@ -128,6 +135,12 @@ def main(print_only=False, check=False):
         lambda: ['There are %d lints included in this crate:\n' % len(lints)],
         write_back=not check)
 
+    # update the `pub mod` list
```
Comment (llogiq) [inline util/update_lints.py]: This is a good idea. Makes me wonder why we didn't do this all the time... :+1:

**Comment by mcarton** [comment]
Done, this time I did not forget half the lint :smile:.

**Comment by llogiq** [comment]
Cool stuff!

**Pull request merged**